pkg/config/autodiscover/autodiscover_generic.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2020 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package autodiscover
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/test-network-function/test-network-function/pkg/config/configsections"
+)
+
+const (
+	genericLabelName = "generic"
+	underTestValue   = "target"
+	// partnerLabelValue = "partner"
+	orchestratorValue          = "orchestrator"
+	fsDiffMasterValue          = "fs_diff_master"
+	skipConnectivityTestsLabel = "skip_connectivity_tests"
+)
+
+// BuildGenericConfig builds a `configsections.TestConfiguration` from the current state of the cluster,
+// using labels and annotations to populate the data.
+func BuildGenericConfig() (conf configsections.TestConfiguration) {
+	var partnerContainers []configsections.Container // PartnerContainers is built from all non-target containers
+
+	// an orchestrator must be identified
+	orchestrator, err := getContainerByLabel(genericLabelName, orchestratorValue)
+	if err != nil {
+		log.Fatalf("failed to identify a single test orchestrator container: %s", err)
```

Comment:
It is. I considered the complexity of attempting to do something else that might not have side effects to be too high at this point. That may change if/when the tool is being used to do much longer-running tests and the cost of a hard stop is therefore higher. At that point it would be useful to allow whole specs to halt in cases like this, but I think that is part of a larger piece to standardise how ginkgo scopes and gomega `Expects` are used throughout.